### PR TITLE
Fix frozen Range issue

### PR DESCRIPTION
**Pull request by ohler55**
In this PR an array is first created and populated during loading. As the array is popped from the stack in `end_element` a Range object is created.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -14,7 +14,8 @@ rvm:
   - 2.5.7
   - 2.6.3
   - 2.6.5
-  - 2.7.0
+  - 2.7.2
```
Comment (voxik) [inline .travis.yml:17]: It would be nice to include `ruby-head` into the testing. That way, similar issues could be cauht earlier.

**Comment by mtasaka** [comment]
Ah, okay, although I don't know well about ox stack structure, perhaps I understand what you want to do, and this seems to make sense.   Also, test suite passes both with `ruby 2.7.2p137` and `ruby 3.0.0p0` . Thank you!

File: .travis.yml (hunk 1 above)
Comment (ohler55) [reply .travis.yml:17]: Sadly travis is alway behind with Ruby releases. It does not yet support Ruby 3.0.0. Tried it and the Ruby 3.0.0 could not be found. It does make sense to use head though.

Comment (ohler55) [reply .travis.yml:17]: Update on that. I see ruby-head is pointing to Ruby 3.1.0 but the image does not include make if I am reading the error message correctly. Will set aside that option for a separate branch. If you feel so inclined to update the .travis.yml file to get it working I'd be glad to review a PR for that.

**Pull request merged**